### ISSUE TITLE
Show price quotes in preferred currency for every trading pair

### ISSUE THREAD

**Issue by calvinfroedge** [bitshares/bitshares-ui#353]
From Luke Stokes via Facebook:

```
Luke Stokes I'd love to see a way to display prices in USD terms. Maybe with a mouse over? It's hard to know if 29,222.86996 BTS per BTC is a good price.

Calvin Froedge Show prices in BITUSD. Go to settings and change your "Preferred Unit of Account" to USD.

Luke Stokes I already have that turned on. I mean on the actual trading pairs. Like if I'm looking at the BTS/BTC exchange, and I see a rate of 29,222.86996 BTS per BTC, I have to then go to my spreadsheet and figure out what that means in terms of USD. A mouse over of the individual buy/sell prices in USD terms (or preferred currency terms) would be _so_ helpful.
```
This would mean that when buying / selling asset pairs, or even looking at the order book, we would be able to see things in terms of our preferred unit of account. Perhaps another column or a hover.

**Comment by wmbutler** [comment]
This is a really important issue IMO. Every person I've introduced to BTS has to have coinmarketcap and a calculator by their side in order to be sure they are setting an order somewhat close to a fair market price. Let's take this example:

## Trade BTS for ETH on the DEX

The Market is trading at 2142 BTS / ETH. In order to determine if this is a fair price, I need to know the average price for ETH and the average price for BTS at that moment (in my preferred currency). If we utilized the coinmarketcap API, we could quickly determine that ETH is trading at 300/USD and that BTS was trading at .14/USD. We could then overlay a fair market value line at 2142. Hovering on this line could display both prices relative to USD and a visual of the calculation used to arrrive at the overlay.

This is one of many potential approaches. It needs continued discussion and UX ultimately. I can guarantee that having ths type of feature would allow BTS users to trade more quickly and with greater confidence.

**Comment by svk31** [comment]
Coinmarketcap does not have an API, but there are websites offering APIs based on their data. The problem then becomes that we rely on external services, which can go down, get abandoned/shut down etc.

I do think showing USD equivalents for example would be useful, similar to how Bittrex does it, but I think we could use the existing price conversion using only internal data. That would however mean that some conversions are not possible due to lack of data, for new or illiquid markets for example.

**Comment by wmbutler** [comment]
Rather than showing USD equivalents, we should show equivalents relative to their preferred currency in settings. coincap.io has a really nice sockets based api that might play well with bitshares.

https://github.com/CoinCapDev/CoinCap.io

**Comment by svk31** [comment]
Actually CMC does have an API now, I hadn't checked in a while: https://coinmarketcap.com/api/

**Comment by btsfav** [comment]
how about adding both with the option to select one?

**Comment by wmbutler** [comment]
Makes more sense to just pick one. @svk31 Do they both have sockets implementation? I think choosing the one with sockets would be ideal for our app.

**Comment by tbone-bts** [comment]
I agree that this would be extremely useful.  But it's not always useful (or preferable) to see the equivalent price in terms of fiat.  In fact, sometimes the user may be looking at a market already priced in fiat (perhaps because it's the most liquid market for the quoted asset in question) but may wish to see the price equivalent in terms of BTC or even BTS.  So it all depend on the situation, and I think this calls for a flexible solution.  

For starters, I agree with whoever suggested the alternate price should be visible via a hover (not as an extra column since space constraints would seem to make this difficult).  As far as how the user can specify which alternate base currency they want to be visible via hover, perhaps there can be a drop down located in some of the empty space below the chart, to the far left of the chart controls.  The drop down could include a) user's prefered fiat currency, b) BTC, c) BTS, and perhaps d) ETH.  Thoughts?

**Comment by wmbutler** [comment]
I can explain what I observe from inexperienced users daily. My buddy Ron wants to trade ETH for LTC. So he goes to CoinMarketCap ETH/USD and determines ETH is trading for $300. Then he goes to CoinMarketCap LTC/USD and determines LTC is trading for $60. Then he pulls out a calculator and determines that a fair price across the broad markets is 5 LTC/ETH. Then he looks at the DEX order book. So you see, the user doesn't really need a preferred currency to make this happen because the presentation (possibly a vertical line on the market) simply depicts a fair trade price based upon a really simple calculation.

If anything, we could just draw on the preferred unit of account setting to be sure that we are running the comparison in the same units, but it's probably safe enough to use the USD against each currency. Remember, it gets canceled out in the calculation.

![screen shot 2017-09-19 at 9 27 13 pm](https://user-images.githubusercontent.com/1254810/30624195-9e3134a8-9d81-11e7-8d16-6bf485fa2da6.png)

**Comment by abitmore** [comment]
Basically I'm against using external API calls. The data may be wrong, then who will be responsible for potential loss? For most popular fiats/crypto currencies we have price feeds.

**Comment by wmbutler** [comment]
It's a convenience to users and I think it's appropriate. The wording is definitely important because we don't want to give people a false sense of security. But it would be a huge help to users who are constantly doing manual calculations to find out if they can place a trade. We could even leave it turned off by default and ensure that users have to check a box on a disclaimer to add it.

**Comment by svk31** [comment]
I've always been against using external APIs as well which is why I never added something like this in the first place. For any active market or bit asset it will be possible to use internal data to provide conversions.

**Comment by btsfav** [comment]
make it optional?

**Comment by wmbutler** [comment]
Definitely optional. I had an idea for including a calculator icon that a user could click in the exchange to optionally view the fair market trade price. I'll UX this.

**Comment by cogutvalera** [comment]
as a good reference of price source look it here https://github.com/Zapata/bitshares-pricefeed/blob/develop/README.md
